rcl/test/CMakeLists.txt

```
@@ -23,6 +23,27 @@ set(extra_lib_dirs "${rcl_lib_dir}")
 set(test_resources_dir_name "resources")
 add_definitions(-DTEST_RESOURCES_DIRECTORY="${CMAKE_CURRENT_BINARY_DIR}/${test_resources_dir_name}")
 
+set(DISTRIBUTION "Unknown")
+if("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux")
+  # If we are on Linux, look for /etc/os-release, which has "key=value" items.
+  # Parse those items, looking for a key named "id" (case-insensitive), and
+  # if we find it, set DISTRIBUTION to that value.  That gives us some idea
+  # of which Linux distribution we are on.
+  if(EXISTS "/etc/os-release")
+    file(STRINGS "/etc/os-release" OS_RELEASE)
+    foreach(line ${OS_RELEASE})
+      string(REGEX REPLACE "^(.*)=.*" "\\1" key "${line}")
+      string(TOLOWER "${key}" key)
+      if("${key}" STREQUAL "id")
+        string(REGEX REPLACE "^.*=(.*)" "\\1" DISTRIBUTION "${line}")
```

@clalancette nit: if we do
```suggestion
        string(REGEX REPLACE "^.*=\"(.*)\"" "\\1" DISTRIBUTION "${line}")
```
we can forget about quotes later on.

---

Sigh.  Unfortunately, on Ubuntu, the quotes are *not* there, so the regex wouldn't match at all.  I guess we could take that as a sign that this is not CentOS, but I think what is there right now is a little more generic (even though we have to worry about the quotes).

---

Interesting. Nevermind then.